tests/basic_test.go

```diff
@@ -518,11 +518,16 @@ func testFailureResolver(t *testing.T) {
 
 func resetTimeStamp(generatedState types.GeneratedState) types.GeneratedState {
 	generatedState.MetaInfo.TimeStamp = time.Time{}
-	for captureID, captureState := range generatedState.Cache.Capture {
+	generatedState.Cache.Capture = resetTimestampAtCapturedStates(generatedState.Cache.Capture)
+	return generatedState
+}
+
+func resetTimestampAtCapturedStates(capturedStates map[string]types.CaptureState) map[string]types.CaptureState {
+	for captureID, captureState := range capturedStates {
 		captureState.MetaInfo.TimeStamp = time.Time{}
```

Comment:
Unrelated really, but is this doing something? Is it mutation a local copy or the original?

Reply:
And then stored at the returned map `capturedStates[captureID] = captureState`.

Reply:
Pfff... confusing.
It could just be direct:
```
for captureID := range capturedStates {
    capturedStates[captureID].MetaInfo.TimeStamp = time.Time{}
```

Reply:
You cannot do that at golang https://go.dev/play/p/ZMmNAAwVOyV